### ISSUE TITLE
[otbn] Merge simulator into master branch

### ISSUE THREAD

**Issue by imphil** [lowRISC/opentitan#2931]
The OTBN simulator has been bootstrapped in the `otbn-dev` branch so far. We need to clean it up as necessary, and get it merged into master.

While at it, we also need to document how to install and use it. This could potentially be done in the same document as #2930.

**Comment by rswarbrick** [comment]
About installation: I'd much prefer not to use `setup.py` for this. The problem is that then you are essentially installing things in the user's environment. Cue massive confusion if there are two OpenTitan repositories on the user's machine. (We'll pick up whichever version of the simulator corresponds to the last call to pip3).

A slightly hacky solution is hacking `sys.path` at the top of scripts, using `__file__` to figure out where library code lives. Maybe there's a nicer way to do things that still doesn't modify stuff outside of the tree? (As far as I'm concerned, writing stuff to `~/.local/lib/python3.8/site-packages/otbnsim.egg-link` is a sign that we're doing it wrong).

**Comment by imphil** [comment]
Well, encouraging people to use a venv would be the recommended solution I guess, but I'm not sure what other can of worms that would open in the opentitan world.

**Comment by rswarbrick** [comment]
Yeah, and I agree that's just not going to happen. My favoured approach is to use relative imports within `hw/ip/otbn` (where we can add a liberal sprinkling of `__init__.py` files) and rely on `sys.path` hacking if we need to go outside of there.

**Comment by imphil** [comment]
@wallento @rswarbrick What's missing to get this one done?

**Comment by wallento** [comment]
We had this pending discussion around venv in #3036. I think the idea was to just not call it properly via the environment but with the relative path in the repo and have the broader discussion in parallel.

**Comment by rswarbrick** [comment]
Yep, that sounds right. I've done that in PR #3199, which contains the first bit of your simulator work (the difficult bit is coping without having the simulator on the path).

**Comment by wallento** [comment]
Awesome, thanks! Well, that difficult bit is what the environment is for, right ;)
